x-pack/plugins/security_solution/public/management/pages/trusted_apps/view/hooks.ts

```diff
@@ -13,10 +15,37 @@ import {
   MANAGEMENT_STORE_GLOBAL_NAMESPACE as GLOBAL_NS,
 } from '../../../common/constants';
 
-import { TrustedAppsListPageState } from '../state';
+import { AppAction } from '../../../../common/store/actions';
+import { getTrustedAppsListPath } from '../../../common/routing';
+import { TrustedAppsListPageLocation, TrustedAppsListPageState } from '../state';
+import { getCurrentLocation } from '../store/selectors';
 
 export function useTrustedAppsSelector<R>(selector: (state: TrustedAppsListPageState) => R): R {
   return useSelector((state: State) =>
     selector(state[GLOBAL_NS][TRUSTED_APPS_NS] as TrustedAppsListPageState)
   );
 }
+
+export type NavigationCallback = (
+  ...args: Parameters<Parameters<typeof useCallback>[0]>
+) => Partial<TrustedAppsListPageLocation>;
+
+export function useTrustedAppsNavigateCallback(callback: NavigationCallback) {
+  const location = useTrustedAppsSelector(getCurrentLocation);
+  const history = useHistory();
+
+  return useCallback(
+    (...args) => history.push(getTrustedAppsListPath({ ...location, ...callback(...args) })),
+    // TODO: needs more investigation, but if callback is in dependencies list memoization will never happen
```

Comment:
Yeah, i'm not seeing value in using this hook. I would propose you revert back to using `useCallback()` within the context of the component